### ISSUE TITLE
OpenGL uses a lot more text shadow

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#16234]
Is it on purpose or an oversight that OpenGL seems to use a lot more _text shadow_ where Version 1 rendering does not, e.g. for village names, mountain captions, mountain ranges. Observed both in default render as well as dependent map styles like Touring view. See here  V1 on the left, v2 on the right:

![Screenshot_20230114-090537_OsmAnd~](https://user-images.githubusercontent.com/845735/212463160-0a61f5fd-eae1-48e9-8c4e-a6a08443b938.jpg) ![Screenshot_20230114-090524_OsmAnd~](https://user-images.githubusercontent.com/845735/212463159-fc2bdfc1-d394-4e36-9d8a-df92b024d93f.jpg)

![Screenshot_20230114-090319_OsmAnd~](https://user-images.githubusercontent.com/845735/212463158-3e4650a0-9daf-4acd-accb-0b6b2666fcf5.jpg) ![Screenshot_20230114-090302_OsmAnd~](https://user-images.githubusercontent.com/845735/212463155-eb504db2-2e6b-4ef0-8378-971ecfa7ad42.jpg)

**Comment by pebogufi** [comment]
V1 isbetter readable

**Comment by sonora** [comment]
PS: It appears that not just text labels use the shadow but even _icons_, like the mountain top icon?

![Screenshot_20230114-092935_OsmAnd~](https://user-images.githubusercontent.com/845735/212463556-7a5797be-0a94-481e-9dc7-6177c02c1d34.jpg)

**Comment by arncor** [comment]
I noticed that too, I prefer the v1 looks too ;)

**Comment by sonora** [comment]
Yes, I prefer V1, too, it should always be our benchmark as it's been tested and tweaked over many years.

PS: And please don't use my screenshots above for the final quality assessment, I have reduced the size of the screenshots to fit them better in this post, they are not the original resolution and hence artificially blurred.

**Comment by vshcherb** [comment]
Comparing opengl / v1 don't see much difference

<img src="https://user-images.githubusercontent.com/1042025/217841651-41aa20ec-f6a0-46de-b888-0acaf3657de3.jpg" width="30%"/> <img src="https://user-images.githubusercontent.com/1042025/217841669-7e8f2825-8e72-4890-9bc8-aefa1a0eb1c8.jpg" width="30%"/>

**Comment by sonora** [comment]
Here are partial screenshots in my original screen resolution (there are some slight jpg artifacts by cutting):

Look village label "_Castelluccio_" and the mountain peak label "_Monte Veletta (1614)_" to spot the difference: 
![1](https://user-images.githubusercontent.com/845735/217868319-cdca57ee-3b7a-4a38-8bff-5dca3383284c.png) ![2](https://user-images.githubusercontent.com/845735/217868349-7c86a271-756b-4785-b807-482a7a160b85.png)

While above it seems we simply use regular letters in V1 and cased letters in V2, also, look at the **road casing** here: Doesn't it look somehow blurred and hence thicker and darker in the bottom (V2) case? (But other tha ncased vs. uncased letters above, this second observation may just be how OpenGL handles pixelation, so this may not be anything to fix.):
![3](https://user-images.githubusercontent.com/845735/217868366-f43b044f-48fb-47d1-9f78-ca354c88789b.png) ![4](https://user-images.githubusercontent.com/845735/217868374-e071d1ec-5fff-4353-8722-bab1321bb67e.png)

**Comment by vshcherb** [comment]
I think I've got what's going on... The problem that v1 puts hillshade on top of the text and it looks less prominent but v2 does it other way around, in some sense it's a matter of taste but this one of the issue which we couldn't fix or choose in v1 or v2, so far. So we even make special setting for Overlay setting to have it half-transparent

**Comment by sonora** [comment]
Yes, that explains it, Victor! I was wondering why also in v1 there seems to be a very faint letter casing if you look vvery close, and what you state sounds like the answer.

Cannot say what a proper fix ccould be and how complex it would be. Sounds like there are opinions that the v1 appearance is superior, but that could also be a matter of what we have become used to.

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14172mqta, Released: 2023-02-12
Indeed, with V2 rendering, the text is more visible and clearer than with V1 rendering. But, as Victor explained above, this is how OpenGL works.
And, yes, it's most likely a matter of taste - I prefer the picture with V2 rendering (if we're talking about a specific example) -  text looks lighter, which in turn is easier to perceive.

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/218300927-ed838fa4-789c-44c3-8abe-67d1805be693.jpeg"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/104760013/218300930-39719535-178c-489b-b00c-9457743115df.jpeg"  width="300" height="auto"/> </th>
  </tr>
</table>
